adapters/targetVideo/targetvideo.go

```diff
@@ -0,0 +1,194 @@
+package targetVideo
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/prebid/openrtb/v20/openrtb2"
+	"github.com/prebid/prebid-server/v3/adapters"
+	"github.com/prebid/prebid-server/v3/config"
+	"github.com/prebid/prebid-server/v3/errortypes"
+	"github.com/prebid/prebid-server/v3/openrtb_ext"
+	"github.com/prebid/prebid-server/v3/util/jsonutil"
+)
+
+type adapter struct {
+	endpoint string
+}
+
+type impExt struct {
+	TargetVideo openrtb_ext.ExtImpTargetVideo `json:"targetVideo"`
+}
+
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	totalImps := len(request.Imp)
+	errors := make([]error, 0)
+	adapterRequests := make([]*adapters.RequestData, 0, totalImps)
+
+	// Split multi-imp request into multiple ad server requests. SRA is currently not recommended.
+	for i := 0; i < totalImps; i++ {
+		if adapterReq, err := a.makeRequest(*request, request.Imp[i]); err == nil {
+			adapterRequests = append(adapterRequests, adapterReq)
+		} else {
+			errors = append(errors, err)
+		}
+	}
+
+	return adapterRequests, errors
+}
+
+func (a *adapter) makeRequest(request openrtb2.BidRequest, imp openrtb2.Imp) (*adapters.RequestData, error) {
+
+	// For now, this adapter sends one imp per request, but we still
+	// iterate over all imps in the request to perform the required
+	// imp.ext transformation.
+	request.Imp = []openrtb2.Imp{imp}
+
+	_, errImp := validateImpAndSetExt(&imp)
+	if errImp != nil {
+		return nil, errImp
+	}
+
+	for i := range request.Imp {
+		if len(request.Imp[i].Ext) == 0 {
+			continue
+		}
```

Comment:
You should be able to delete this as this should never be true given the logic in `validateImpAndSetExt` and that your adapter will never be called with an empty imp.ext.